recipes/leftbar/general_row/general_row.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+  <li
+    :class="leftbarGeneralRowClasses"
+    data-qa="leftbar-general-row"
+  >
+    <div class="d-d-flex d-ai-center d-fl-grow2 d-truncate">
+      <div class="d-w24 d-wmn24 d-h24 d-d-flex d-ai-center d-jc-center">
+        <dt-recipe-leftbar-general-row-icon
+          :type="getIcon"
+          :color="color"
+          class="leftbar-general-row__icon"
+        />
+      </div>
+      <dt-emoji-text-wrapper
+        :class="descriptionClasses"
+        data-qa="leftbar-general-row-description"
+      >
+        {{ description }}
+      </dt-emoji-text-wrapper>
+    </div>
+    <div class="d-d-flex d-ai-center d-pl8">
+      <dt-badge
+        v-if="hasUnreadMessages"
+        kind="count"
+        color="purple-400"
+        type="bulletin"
+        data-qa="leftbar-general-row-unread-badge"
+      >
+        {{ unreadCount }}
+      </dt-badge>
+    </div>
+  </li>
+</template>
+
+<script>
+import { LEFTBAR_GENERAL_ROW_TYPES, LEFTBAR_GENERAL_ROW_CONTACT_CENTER_COLORS } from './general_row_constants.js';
+import { DtBadge } from '@/components/badge';
+import DtEmojiTextWrapper from '@/components/emoji_text_wrapper/emoji_text_wrapper.vue';
+import DtRecipeLeftbarGeneralRowIcon from './leftbar_general_row_icon.vue';
+
+export default {
+  name: 'DtRecipeGeneralRow',
+
+  components: {
+    DtEmojiTextWrapper,
+    DtBadge,
+    DtRecipeLeftbarGeneralRowIcon,
+  },
+
+  props: {
+    /**
+     * Determines the icon to show.
+     * If type is contact center, the color prop must be provided and will determine the color of the icon
+     */
+    type: {
+      type: String,
+      default: 'inbox',
+      validator: (type) => {
+        return Object.values(LEFTBAR_GENERAL_ROW_TYPES).includes(type);
+      },
+    },
+
+    /**
+     * Text displayed next to the icon
+     */
+    description: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * Determines the color of the contact center icon
+     */
+    color: {
+      type: String,
+      default: null,
+      validator: (color) => {
+        return Object.keys(LEFTBAR_GENERAL_ROW_CONTACT_CENTER_COLORS).includes(color);
+      },
+    },
+
+    /**
+     * Number of unread messages
+     */
+    unreadCount: {
+      type: Number,
+      default: null,
+    },
+  },
+
+  computed: {
+    leftbarGeneralRowClasses () {
+      return [
+        'leftbar-general-row',
+      ];
+    },
+
+    descriptionClasses () {
+      return [
+        'leftbar-general-row__description',
+        'd-mr-auto',
+        'd-truncate',
```

Comment:
Ellipsis isn't appearing

<img width="407" alt="image" src="https://user-images.githubusercontent.com/1165933/219549222-9ed7ede1-e273-485d-aea8-562b7547989b.png">

Reply:
`text-overflow:ellipsis;` doesn't appear to work with `display: flex` @josedialpad can you remove `display: flex` from `leftbar-general-row__description` or will this cause problems with the UI?

Reply:
Removed the display flex from the description https://github.com/dialpad/dialtone-vue/pull/769/commits/4c5fe4fa070767b7eebe9e29ae10f490c3f152d7